### PR TITLE
Bugfixes: Respect visibility of roles + "Moderator by default" setting was broken

### PR DESCRIPTION
**Pull request by abias**
Please see the commit messages for details

### CODE REVIEW

File: locallib.php [hunk 1]
```diff
@@ -687,29 +687,51 @@ function($u) {
...
  * @return array $roles
  */
-function bigbluebuttonbn_get_roles(context $context = null) {
-    $roles = (array) role_get_names($context);
-    foreach ($roles as $key => $value) {
-        $roles[$key] = $value->localname;
+function bigbluebuttonbn_get_roles(context $context = null, bool $onlyviewableroles = true) {
+    global $CFG;
+
+    if ($onlyviewableroles == true && $CFG->branch >= 35) {
```
Comment (jfederico) [inline locallib.php:697]: What is that hardcoded branch set to 35?

Comment (abias) [reply locallib.php:697]: The get_viewable_roles() function was added with MDL-50666 in Moodle 3.5.
That's why the fix of this patch cannot be applied to Moodle core versions before 3.5.
As your plugin releases are covering multiple Moodle core versions, this is the only viable solution.

**Comment by abias** [comment]
Hi @jfederico ,

thanks for your feedback.

> The problem you are trying to solve

This patch solves two problems:

1. This plugin does not respect the visibility of roles.

On /admin/roles/allow.php?mode=view, an admin can configure which roles as visible to other roles. This is especially helpful if the admin has created additional roles in the system for special permissions which should not be known to all users of the system. This can also be helpful if the admin simply wants to shorten the list of roles available to the end users and to remove roles which the end user simply does not have to deal with but which also can't be removed from the system completely for any reason.

Unfortunately, plugins which have features to show a list of roles as the BBB plugin does and which do not respect the visibility of roles might still leak these invisible roles to users.

2. The "Moderator by default" setting was broken. This should be quite obvious. I did not create a separate PR for that as the solution overlaps with the solution for problem no.1 

> The way to reproduce it

Steps to replicate problem no. 1:

* Login as admin
* Go to /admin/roles/allow.php?mode=view
* Uncheck the checkbox in the matrix to prevent the editing teacher role from seeing the manager role
* Login as editing teacher
* Go to a course
* Add a BBB activity

Expected results:
* The list of roles must not contain the "Manager" role

Actual result:
* The list of roles does contain the "Manager" role

**Comment by jfederico** [comment]
Thank @abias I'll check it out today

**Comment by abias** [comment]
@jfederico Do you have any news about this PR or is there anything I can do to help you to push this forward?

**Pull request merged**